game-core/src/main/java/games/strategy/engine/chat/PlayerChatRenderer.java

```diff
@@ -54,9 +52,7 @@ public Component getListCellRendererComponent(final JList<?> list, final Object
       final StringBuilder sb = new StringBuilder(((INode) value).getName());
       final Set<String> players = playerMap.get(value.toString());
       if (players != null && !players.isEmpty()) {
-        sb.append(" (");
-        sb.append(Joiner.on(", ").join(players));
-        sb.append(")");
+        sb.append(players.stream().collect(Collectors.joining(", ", " (", ")")));
```

Comment:
The string builder is now uneccessary IMO because the only "optional addition" is this line here, so if this functionality was ectracted into a helper method it would either return the node name or the node name + the players

Reply:
2998db052 I don't think my brain is firing on all cylinders today.  I couldn't see how to get rid of the `StringBuilder`.  Did I misunderstand your suggestion?

Reply:
Not quite what I intended to say, no.
I was thinking of something like this (also only affecting the else branch):
```java
if (icons != null) {
  // code in if statement
} else {
  super.getListCellRendererComponent(list, getNodeLabelWithPlayers(node), index, isSelected, cellHasFocus);
}
```
where getNodeLabelWithPlayers is something like this:
```java
private static String getNodeLabelWithPlayers(INode node) {
  final Set<String> playerNames = playerMap.getOrDefault(node.toString(), Collections.emptySet());
  if (!playerNames.isEmpty()) {
    return node.getName() + playerNames.stream().collect(Collectors.joining(", ", " (", ")"));
  }
  return node.getName();
}
```

Reply:
Thanks for the clarification. 2e1d62eab